rfc/mediachain-rfc-1.md

```diff
@@ -0,0 +1,435 @@
+ }
+ "_2" : ...
+ ...
+ }
+
+IPRSSignature = {
+ "value" : <Reference>
+ }
+
+```
+
+### Chains and Links
+
+The fundamental operation of the Mediachain is to link entities and
+artefacts with labeled relationships and metadata.  The data model
+described so far is disconnected: it contains independent entities and
+artefacts. In order to track the evolving relationships between
+entities and artefacts we need to construct chains of updates for each
+entity and artefact in the datastore.
+
+The natural way to model chains in a distributed append-only store is
+through linked lists. Thus for each entity and artefact in the
+system, we also have an associated chain which initially points to the
+canonical of a global `Nil` object:
+```
+Chain = <ChainCell> | <Nil>
+ChainCell = {
+ "type" : ...
+ "chain" : <Reference>
+ <Key> : <Value> ...
+ }
+Nil = { "type" : "nil" }
+```
+
+The entity and artefact chains differ on the type of relationships
+they store. They both support metadata update cells, which simply
+provide new metadata to the base object as arbitray key-value pairs.
+
+The entity chain also allows for linking two entities, for example
+having a pseudonymous artist publicly revealing a true name that
+points to another entity in the system.
+
+The artefact chain further stores links to entities that
+represent creation and potential Intellectual Property rights,
+references in the Internet media-space, and derivative relations
+to other artefacts.
+
+Each chain cell carries also carries one or more signatures by
+signatory entities in the system.
+
+Note that the types of cells that can appear in either type of
+chain it is not limited to the types of relationships described
+above. These are merely the minimum schema to support the
+functionality of the Mediachain.
+
+Keeping this in mind, we can model the chains with the following
+schema:
+```
+EntityChainCell =
+   <EntityUpdateCell>
+ | <EntityLinkCell>
+ | <Nil>
+
+EntityUpdateCell = {
+ "type" : "entityUpdate"
+ "chain" : <Reference>
+ "entity" : <Reference>
+ "signatures" : <Signatures>
+  <Key> : <Value> ... ; metadata updates
+ }
+
+EntityLinkCell = {
+ "type" : "entityLink"
+ "chain" : <Reference>
+ "entity" : <Reference>
+ "entityLink" : <Reference>
+ ["relationship" : <Reference>]
+ "signatures" : <Signatures>
+ <Key> : <Value> ... ; entity relationship metadata
+ }
+
+ArtefactChainCell =
+   <ArtefactUpdateCell>
+ | <ArtefactCreationCell>
+ | <ArtefactDerivationCell>
+ | <ArtefactOwnershipCell>
+ | <ArtefactReferenceCell>
+ | <Nil>
+ 
+ArtefactUpdateCell = {
+ "type" : "artefactUpdate"
+ "chain" : <Reference>
+ "artefact" : <Reference>
+ "signatures" : <Signatures>
+ <Key> : <Value> ... ; metadata updates
+ }
+
+ArtefactCreationCell = {
+ "type" : "artefactCreatedBy"
+ "chain" : <Reference>
+ "artefact" : <Reference>
+ "entity" : <Reference>
+ "signatures" : <Signatures>
+ <Key> : <Value> ... ; creation metadata
+ }
+
+ArtefactDerivationCell = {
```

Comment:
We can revisit specific cell once we have implemented enough of a spec to know how to tackle this.
We can always attach more specified fields based on how we crack this issue, it could contain a fingerprint from a suitable fingerprinting procedure depending on the media type.